### PR TITLE
Extraction of the inline documentation into a separate file

### PR DESCRIPTION
**Pull request by ph**
For details see https://github.com/elastic/logstash/issues/6846

### CODE REVIEW

File: docs/index.asciidoc [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+This plugin supports the following configuration options:
+
+* <<plugins-{type}s-{plugin}-options>>
+* <<plugins-{type}s-common-options>>
+
+[id="plugins-{type}s-{plugin}-options"]
+==== Dissect Filter Options
+
+The following configuration options are supported by the dissect filter:
```
Comment (dedemorton) [inline docs/index.asciidoc]: @ph I'm really sorry to bring this up now (should have noticed it earlier), but when I look at the long list of options for some plugins, I'm concerned that people might overlook the common options (yes, even though there is a link right above this...people don't read carefully). What if we were to remove line 160/161 and change line 150 to say:

`This plugin supports the following configuration options plus the <<plugins-{type}s-common-options>> described later.`

Comment (dedemorton) [inline docs/index.asciidoc]: @ph In fact, if you implement my suggestion, you could remove lines 140 through 145 and just change the title  on line 148 to "Dissect Filter Configuration Options".

Comment (ph) [inline docs/index.asciidoc]: I have fixed it in the `template` for the ruby content in https://github.com/elastic/logstash/pull/6941/commits/b32e5cf3acb16d9e017fe80f449bdbdc0eac8892

**Comment by ph** [comment]
DO NOT MERGE

**Pull request closed without merging**